File: php7/lib/Options.php
<?php
/* -*- coding: utf-8 -*-
 *
 * jomiel-examples
 *
 * Copyright
 *  2019 Toni Gündoğdu
 *
 *
 * SPDX-License-Identifier: Apache-2.0
 */

namespace Demo;

use GetOptionKit\OptionPrinter\ConsoleOptionPrinter;
use GetOptionKit\OptionCollection;
use GetOptionKit\OptionParser;

function opts_init($argv) {
    $opts = new OptionCollection;

    $opts->add("h|help", "Print help and exit");

    $opts->add("V|version-zmq", "Display ZeroMQ version and exit");

    $opts->add("r|router-endpoint:",
        "Specify the rounter endpoint address")
        ->isa("string")
        ->defaultValue("tcp://localhost:5570")
        ;

    $opts->add("t|connect-timeout:",
        "Specify maximum time in seconds for the connection allowed to take")
        ->isa("number")
        ->defaultValue(30)
        ;

    $opts->add("q|be-terse",
        "Be brief and to the point; dump interesting details only");

    $parser = new OptionParser($opts);
    try {
        $result = $parser->parse($argv);
        // --help
        if ($result->help) {
            $printer = new ConsoleOptionPrinter;
            echo "Usage:\n";
            echo "    demo [options] <uri> ...\n\n";
            echo $printer->render($opts);
            die;
        // --version-zmq
        } else if ($result->versionZmq) {
            echo "ZeroMQ version " . ZMQ::LIBZMQ_VERSION . " ("
                . ZMQ::LIBZMQ_VERSION_ID . ")\n";
            die;
        }
        $args = $result->getArguments();
    } catch (Exception $e) {
        echo $e->getMessage();
        die;
    }
    return $result;
}

?>

File: lua5/src/demo/options.lua
--
-- -*- coding: utf-8 -*-
--
-- jomiel-examples
--
-- Copyright
--  2019-2021 Toni Gündoğdu
--
--
-- SPDX-License-Identifier: Apache-2.0
--

local Options = {}

function Options.parse()
  local Parser = require 'argparse' 'demo'

  Parser:flag('-D --print-config',
    'Print confuguration values and exit')

  Parser:flag('-V --version-zmq',
    'Display ZeroMQ version and exit', false)

  Parser:flag('-j --output-json',
    'Print dumped messages in JSON', false)

  Parser:flag('-c --compact-json',
    'Use more compact representation of JSON', false)

  Parser:option('-r --router-endpoint',
    'Specify the outer endpoint address',
    'tcp://localhost:5514')

  Parser:option('-t --connect-timeout',
    'Specify maximum time in seconds for the connection allowed to take',
    30)

  Parser:flag('-q --be-terse',
    'Be brief and to the point; dump interesting details only', false)

  Parser:option('-p --proto-dir',
    'Specify path to the dir with the .proto files',
    '../proto'
  )

  Parser:argument('URI'):args('*')

  return Parser:parse()
end

return Options
